include/decoder.hpp

```diff
@@ -7,14 +7,15 @@
  *   Tassilo Tanneberger
  */
 
-#ifndef TETRA_DECODER_DECODER_HPP
-#define TETRA_DECODER_DECODER_HPP
+#pragma once
 
 #include <memory>
 #include <optional>
 #include <string>
 #include <vector>
 
+#include <bit_stream_decoder.hpp>
```

Comment:
why not `#include "bit_stream_decoder.hpp"` ?

I think this is misleading and breaks if the cmake statement is removed.